gcip/addons/security/sops.py

```diff
@@ -0,0 +1,22 @@
+__author__ = "Daniel von Eßen"
+__copyright__ = "Copyright 2020 DB Systel GmbH"
+__credits__ = ["Daniel von Eßen", "Thomas Steinbach"]
+# SPDX-License-Identifier: Apache-2.0
+__license__ = "Apache-2.0"
+__maintainer__ = "Thomas Steinbach"
+__email__ = "daniel.von-essen@deutschebahn.com"
+
+
+def sops_export_decrypted_values(path: str) -> str:
```

Comment:
I think it is more secure and "sops native" to write a function that will take the path of the sops file to encrypt as well as a command to execute and wrapt that command into the `sops exec-env` command:
https://github.com/mozilla/sops/blob/master/README.rst#218passing-secrets-to-other-processes

Reply:
sth. like

```
def sops_exec_env(path: str, command: str) -> str:
    return f"sops exec-env {path} '{command}‘"
```

Reply:
That would be awsome. I think that could be a second function. 
My intention is, to export variables in predefined jobs. E.g. DockerClientConfig needs `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` env vars.
`DockerClientConfig` is used in preconfigured jobs like `crane.push` or `crane.pull`. 
To decrypt and export env vars, I want to just have to `decrypt` and `exoprt` those variables. 

_sops/secrets.env_
```env
REGISTRY_USERNAME=myusername
REGISTRY_PASSWORD=secretpassword
``` 
_gitlab-ci.py_

```python
pipeline = Pipeline()
crane_job = crane.push("my_registry_url", "image", "v1.1.1") # crane_job uses default DockerClientConfig
crane_job.prepend_scripts(sops_export_decrypted_values("sops/secrets.env"))
pipeline.add_children(crane_job)
pipeline.write()  
```
_generated_pipeline.yaml_
```yaml
job:
  script:
    - export $(sops -d sops/secrets.env)
    - mkdir -p $HOME/.docker
    - 'echo "{\"auths\": {\"index.docker.com\": {\"username\": \"$REGISTRY_USERNAME\",
  \"password\": \"$REGISTRY_PASSWORD\"}}}" > $HOME/.docker/config.json'
    - crane push my_registry_url/image:v1.1.1
```

Reply:
check